extensions/vscode/src/views/homeView.ts

```diff
@@ -137,10 +142,24 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
       );
     });
 
+    useBus().on("activeContentRecordChanged", (contentRecord) => {
+      this.contentRecordWatchers?.dispose();
+
+      this.contentRecordWatchers = new ContentRecordWatcherManager(
+        contentRecord,
+      );
+
+      this.contentRecordWatchers.contentRecord?.onDidChange(
+        this.updateServerEnvironment,
+        this,
+      );
+    });
+
     useBus().on(
       "activeConfigChanged",
       (cfg: Configuration | ConfigurationError | undefined) => {
         this.sendRefreshedFilesLists();
+        this.updateServerEnvironment();
```

Comment:
I did not end up getting to the only-update-on-config's-server-attribute feature. I figure I can re-visit this, but if we don't want to take on that technical debt I can address it in this PR.

Reply:
I'm ok with this as is and wouldn't even consider it tech debt unless it had enough impact on performance. IMO this fulfills the goal of secrets syncing nicely without trying to include a more complex solution with extra checks that could expand edge cases.

Reply:
Sounds great! I'm glad I didn't delve into that rabbit hole then